app/utils/service_client/middleware/request_id.rb

```diff
@@ -0,0 +1,11 @@
+module ServiceClient
+  module Middleware
+    class RequestID < MiddlewareBase
+      def enter(ctx)
+        headers = ctx.fetch(:req).fetch(:headers)
```

Comment:
There are couple benefits:

1. If `req` is `nil`, we get more informative error message: `KeyError: key not found: :req` vs `NoMethodError: undefined method `[]' for nil:NilClass`
2. If `headers` is `nil`, we get more informative error message: ``KeyError: key not found: :headers` vs `NoMethodError: undefined method `merge' for nil:NilClass`

It's true that the method is so shorts that even a bad error message would point easily to the issue, but does fetch do some harm?